### PR TITLE
TF-3047 Improve download video on mobile

### PR DESCRIPTION
**Pull request by dab246**
## Issue

#3047 

## Media Type Supported

- Video, Image

## Resolved

- Android

https://github.com/user-attachments/assets/448b1817-e81c-430f-824e-b773684055fa

- iOS

https://github.com/user-attachments/assets/026890d5-334b-4fce-86b9-7aed28dae50e

### CODE REVIEW

File: lib/main/permissions/permission_dialog.dart [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      child: InkWell(
+        borderRadius: BorderRadius.circular(100.0),
+        onTap: onPressed,
+        child: Padding(
+          padding: const EdgeInsets.symmetric(horizontal: 12.0, vertical: 10.0),
+          child: Text(
+            text,
+            style: Theme.of(context).textTheme.labelLarge?.copyWith(
+                  color: Theme.of(context).colorScheme.primary,
+                ),
```
Comment (tddang-linagora) [inline lib/main/permissions/permission_dialog.dart:133]: Indent

File: lib/main/permissions/permission_dialog.dart [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+                    _PermissionTextButton(
+                      context: context,
+                      text: AppLocalizations.of(context).allow,
+                      onPressed: () async {
+                        if (widget.onAcceptButton != null) {
+                          widget.onAcceptButton!.call();
+                        } else {
+                          await widget.permission.request().then(
+                                (value) => Navigator.of(context).pop(),
+                              );
```
Comment (tddang-linagora) [inline lib/main/permissions/permission_dialog.dart:94]: Indent

File: lib/features/base/mixin/save_media_to_gallery_mixin.dart [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+   log('SaveMediaToGalleryMixin::saveImageToGallery:file path: ${file.path}');
+    await Gal.putImage(file.path);
+  }
+
+  Future<void> saveVideoToGallery({
+    required File file,
+  }) async {
+    log('SaveMediaToGalleryMixin::saveVideoToGallery:file path: ${file.path}');
+    await Gal.putVideo(file.path);
+  }
```
Comment (tddang-linagora) [inline lib/features/base/mixin/save_media_to_gallery_mixin.dart:102]: Consider making these methods private. We should only expose methods that are going to be called publicly.

**Comment by hoangdat** [comment]
- [ ] export/save dialog in iOS not have round border in the bottom
Screenshot 2024-09-14 at 05 52 11

File: ios/Podfile [hunk 4]
```diff
@@ -65,7 +65,7 @@ post_install do |installer|
           'PERMISSION_SPEECH_RECOGNIZER=0',
 
           ## dart: PermissionGroup.photos
-          # 'PERMISSION_PHOTOS=0',
+          'PERMISSION_PHOTOS=1',
```
Comment (hoangdat) [inline ios/Podfile:68]: do we need to add more information in `Info.plist`?

File: lib/features/email/presentation/controller/single_email_controller.dart [hunk 5]
```diff
@@ -779,12 +818,43 @@ class SingleEmailController extends BaseController with AppLoaderMixin {
...
+          }
+        );
+      } else {
+        await _openDownloadedPreviewWorkGroupDocument(success.downloadedResponse);
+      }
+    }
   }
 
-  void _openDownloadedPreviewWorkGroupDocument(DownloadedResponse downloadedResponse) async {
+  Future<void> _openDownloadedPreviewWorkGroupDocument(DownloadedResponse downloadedResponse) async {
```
Comment (hoangdat) [inline lib/features/email/presentation/controller/single_email_controller.dart:857]:
```suggestion
  Future<void> _openDownloadedDocument(DownloadedResponse downloadedResponse) async {
```
:)

File: ios/Podfile (hunk 4 above)
Comment (dab246) [reply ios/Podfile:68]: It was declared before, but we will update it correctly to support videos as well.

Screenshot 2024-09-16 at 09 39 09

File: lib/main/permissions/permission_dialog.dart (hunk 2 above)
Comment (dab246) [reply lib/main/permissions/permission_dialog.dart]: done

File: lib/main/permissions/permission_dialog.dart (hunk 1 above)
Comment (dab246) [reply lib/main/permissions/permission_dialog.dart]: done

File: lib/features/base/mixin/save_media_to_gallery_mixin.dart (hunk 3 above)
Comment (dab246) [reply lib/features/base/mixin/save_media_to_gallery_mixin.dart]: done

File: lib/features/email/presentation/controller/single_email_controller.dart (hunk 5 above)
Comment (dab246) [reply lib/features/email/presentation/controller/single_email_controller.dart]: done

**Comment by dab246** [comment]
> * [x]  export/save dialog in iOS not have round border in the bottom
> 
> Screenshot 2024-09-14 at 05 52 11

Fixed. Please see below video demo:


https://github.com/user-attachments/assets/5f266391-c4c8-4822-b275-98592efbe07c

**Review by tddang-linagora — APPROVED** [review]

**Comment by hoangdat** [comment]
- [ ] should have better alignment for this dialog.
image
- [ ] rebase

**Comment by dab246** [comment]
> * [x]  should have better alignment for this dialog.
> 
> image
> * [x]  rebase

Fixed


https://github.com/user-attachments/assets/9b3311bc-5667-457c-9627-3e8684e0f9aa

**Review by hoangdat — APPROVED** [review]

**Pull request merged**